src/main/java/org/apache/commons/imaging/formats/jpeg/JpegImageParser.java

```diff
@@ -598,12 +598,12 @@ public JpegPhotoshopMetadata getPhotoshopMetadata(final ByteSource byteSource,
             final App13Segment segment = (App13Segment) s;
 
             final PhotoshopApp13Data data = segment.parsePhotoshopSegment(params);
-            if (data != null && photoshopApp13Data != null) {
-                throw new ImageReadException(
-                        "Jpeg contains more than one Photoshop App13 segment.");
+            if (data != null) {
+                if (photoshopApp13Data != null) {
+                    throw new ImageReadException("Jpeg contains more than one Photoshop App13 segment.");
+                }
+                photoshopApp13Data = data;
```

Comment:
The logic removed had an issue, where if the first data was not null, then the second was null; the second value would be used. Resulting in missing photoshop APP13 data.